carrent/carrentapp/models.py

```diff
@@ -1,3 +1,126 @@
 from django.db import models
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager, PermissionsMixin
 
 # Create your models here.
+
+class AccountManagerCustom(BaseUserManager):
+
+    def create_user(self,
+                    email,
+                    username,
+                    first_name,
+                    last_name,
+                    birthdate,
+                    addr_city,
+                    addr_street,
+                    addr_post_code,
+                    mobile_nr,
+                    password=None):
+        if not email:
+            raise ValueError("User must have an email address")
+        if not username:
+            raise ValueError("User must have a username")
+        if not first_name:
+            raise ValueError("User must have a first name")
+        if not last_name:
+            raise ValueError("User must have a last name")
+        if not birthdate:
+            raise ValueError("User must have a birthdate")
+        if not addr_city:
+            raise ValueError("User must have a city")
+        if not addr_street:
+            raise ValueError("User must have a street")
+        if not addr_post_code:
+            raise ValueError("User must have a post code")
+        if not mobile_nr:
+            raise ValueError("User must have a phone number")
+
+        user = self.model(
+            email=self.normalize_email(email),
+            username=username,
+            first_name=first_name,
+            last_name=last_name,
+            birthdate=birthdate,
+            addr_city=addr_city,
+            addr_street=addr_street,
+            addr_post_code=addr_post_code,
+            mobile_nr=mobile_nr
+            )
+
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self,
+                         email,
+                         username,
+                         first_name,
+                         last_name,
+                         birthdate,
+                         addr_city,
+                         addr_street,
+                         addr_post_code,
+                         mobile_nr,
+                         password):
+
+        user = self.model(
+            email=self.normalize_email(email),
+            username=username,
+            first_name=first_name,
+            last_name=last_name,
+            birthdate=birthdate,
+            addr_city=addr_city,
+            addr_street=addr_street,
+            addr_post_code=addr_post_code,
+            mobile_nr=mobile_nr)
+
+        user.is_admin = True
+        user.is_staff = True
+        user.is_superuser = True
+
+        user.set_password(password)
+        user.save(using=self.db)
+        return user
+
+
+class UserCustom(AbstractBaseUser):
```

Comment:
```suggestion
class User(AbstractBaseUser):
```
Gdyby jakieś dwa byty Usera występowały to bym dawał jakieś przedrostki żeby rozróżnić, ale zdaje mi się, że po prostu zamieniasz

Reply:
Jeszcze tutaj pytanie, czy zastanawiałeś się nad wyborem `AbstracUser` vs. `AbstractBaseUser`? Myślę, że warto najpierw to przemyśleć, może zaoszczędzić sporo linijek

Reply:
AD UserCustom->User:
Nie wiem czy to przez poprzednie migracje czy coś innego zle robie ale gdy zmieniłem na samo User to sa bledy typu "could not resolve carrentapp.User" (zmieniłem odpowiednio ścierzke w settingsach i  nazwe w powiazanych elementach)

Reply:
Generalnie czym się różni `AbstractUser` od `AbstractBaseUser`, no tym, że Base jest "goły" i prawie wszystko trzeba dodać samemu, a `AbstractUser` już ma to co klasa `User` i można dodać/zmienić tylko to co chcesz

Co do błędu to musiałbym zobaczyć jak to wygląda w projekcie i sobie odpalić bo teraz to ciężko mi powiedzieć, gdzie błąd